containers/gateway/deploy.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash -e
+
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+USAGE="USAGE: ${0} <PROJECT> <ZONE>
+
+Alternatively, you can set the default project and zone using the gcloud tool:
+
+    gcloud config set project <PROJECT>
+    gcloud config set compute/zone <ZONE>
+"
+
+ERR_USAGE=1
+ERR_DOCKER_BUILD=2
+ERR_DOCKER_PUSH=3
+ERR_NETWORK_CREATE=4
+ERR_FIREWALL_RULE=5
+ERR_INSTANCE_CREATE=6
+
+DOCS="This script deploys the Datalab kernel gateway to a GCE VM.
+
+This will:
+
+1. Build the gateway image and push it to GCR.
+2. Ensure that the project contains a network named
+   'datalab-kernels' with inbound SSH connections allowed
+3. Crate a new VM in the default zone connected to the
+   'datalab-kernels' network.
+4. Run the gateway image in that VM
+
+The resulting VM can be used by running the 'run-with-gce.sh'
+script in the 'containers/datalab' directory to set up an
+instance of Datalab connected to that VM via an SSH tunnel.
+"
+
+PROJECT=${1:-`gcloud config list 2> /dev/null | grep 'project = ' | cut -d ' ' -f 3`}
+ZONE=${2:-`gcloud config list 2> /dev/null | grep 'zone = ' | cut -d ' ' -f 3`}
+
+if [[ -z "${PROJECT}" || -z "${ZONE}" ]]; then
+  echo "${USAGE}"
+  exit ${ERR_USAGE}
+fi
+
+echo "${DOCS}"
+
+echo "Will deploy a GCE VM to project ${PROJECT} in zone ${ZONE}"
+read -p "Proceed? [y/N] " PROCEED
+
+if [[ "${PROCEED}" != "y" ]]; then
+  echo "Deploy cancelled"
+  exit
+fi
+
+./build.sh || exit ${ERR_DOCKER_BUILD}
+IMAGE="gcr.io/${PROJECT}/datalab-gateway"
+docker tag -f datalab-gateway "${IMAGE}"
+gcloud docker push "${IMAGE}" || exit ${ERR_DOCKER_PUSH}
```

Comment:
It seems this script is used in development environment, where it relies on the building system. Maybe we should have another "consumer script" which relies on gcloud only? That should be a separate change though.